tests/run-make/macos-deployment-target/rmake.rs

```diff
@@ -0,0 +1,30 @@
+// Check that a set deployment target actually makes it to the linker.
+// This is important since its a compatibility hazard. The linker will
+// generate load commands differently based on what minimum OS it can assume.
+// See https://github.com/rust-lang/rust/pull/105123
+
+//@ only-macos
+// Reason: this test exercises an OSX-specific issue
+
+use run_make_support::{cmd, rustc};
+
+fn main() {
+    rustc()
+        .env("MACOSX_DEPLOYMENT_TARGET", "10.13")
+        .input("with_deployment_target.rs")
+        .output("with_deployment_target.dylib")
+        .run();
+    let pattern = if cmd("uname").arg("-m").run().stdout_utf8().contains("arm64") {
+        "minos 11.0"
+    } else {
+        "version 10.13"
+    };
+    // XXX: The check is for either the x86_64 minimum OR the aarch64 minimum
```

Comment:
Maybe change this to `note` if CI is now unhappy. Dunno why I used `XXX` before.